### ISSUE TITLE
[dev.icinga.com #5810] add check_source_object_id with reference to the endpoint the checker is running on

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#1332]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/5810

**Created by mfriedrich on 2014-03-20 16:54:00 +00:00**

Assignee: _(none)_
Status: _Rejected_  (closed on _2014-03-21 09:16:35 +00:00_)
Target Version: _(none)_
Last Update: _2014-09-16 09:19:42 +00:00 (in Redmine)_


---

That way we may easily join servicestatus with endpointstatus.

    mysql> select oss.name1 as host, oss.name2 as service, ss.state_type as state, ss.output as output, es.is_connected as connected from icinga_servicestatus ss join icinga_objects oss on ss.service_object_id=oss.object_id join icinga_endpointstatus es on ss.check_source_object_id=es.endpoint_object_id join icinga_objects oes on es.endpoint_object_id=oes.object_id;
    +-----------+-----------+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
    | host      | service   | state | output                                                                                                                                                                                                                                                                                                                                                                                                                                    | connected |
    +-----------+-----------+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
    | icinga2b  | cluster   |     1 | Icinga 2 Cluster is running: Connected Endpoints: 1 (icinga2a).                                                                                                                                                                                                                                                                                                                                                                           |         1 |
    | localhost | disk      |     1 | DISK WARNING - free space: / 8442 MB (35% inode=66%): /dev 10 MB (100% inode=99%): /run 793 MB (99% inode=99%): /run/lock 5 MB (100% inode=99%): /run/shm 1587 MB (99% inode=99%): /backup 25544 MB (96% inode=99%): /boot 34 MB (35% inode=98%): /data 19111 MB (43% inode=97%): /home 20891 MB (43% inode=93%): /vms 21991 MB (22% inode=99%): /sys/fs/cgroup 0 MB (100% inode=99%): /home/michi/share/netways 233260 MB (18% inode=-): |         1 |
    | localhost | http      |     1 | HTTP OK: HTTP/1.1 200 OK - 739 bytes in 0,001 second response time                                                                                                                                                                                                                                                                                                                                                                        |         1 |
    | localhost | icinga    |     1 | Icinga 2 is running.                                                                                                                                                                                                                                                                                                                                                                                                                      |         1 |
    | localhost | load      |     1 | OK - load average: 0.27, 0.44, 0.47                                                                                                                                                                                                                                                                                                                                                                                                       |         1 |
    | localhost | ping4     |     1 | PING OK - Packet loss = 0%, RTA = 0.04 ms                                                                                                                                                                                                                                                                                                                                                                                                 |         1 |
    | localhost | ping6     |     1 | PING OK - Packet loss = 0%, RTA = 0.04 ms                                                                                                                                                                                                                                                                                                                                                                                                 |         1 |
    | localhost | processes |     1 | PROCS WARNING: 273 processes                                                                                                                                                                                                                                                                                                                                                                                                              |         1 |
    | localhost | ssh       |     1 | SSH OK - OpenSSH_6.5p1 Debian-6 (protocol 2.0)                                                                                                                                                                                                                                                                                                                                                                                            |         1 |
    | localhost | users     |     1 | USERS OK - 6 users currently logged in                                                                                                                                                                                                                                                                                                                                                                                                    |         1 |
    +-----------+-----------+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
    10 rows in set (0.00 sec)



**Changesets**

_2014-03-20 17:03:47 +00:00_ by _(unknown)_ a26df23e3d55abee4d0bd3c34da49d63702d209f

    DB IDO: Add check_source_object_id reference to endpoint objects.
    
    Refs #5810
    Refs #5636

_2014-03-20 18:34:05 +00:00_ by _(unknown)_ fee925720db35e643ce235530d892c6675c3f4be

    Merge branch 'feature/db-endpoints-5636' into next
    
    Fixes #5636
    Fixes #5690
    Fixes #5810
    Fixes #5811
    Fixes #5812

_2014-03-21 09:18:03 +00:00_ by _(unknown)_ bda7e9f362de2c84dd152a8dbcb46e0de70b2d67

    Revert "DB IDO: Add check_source_object_id reference to endpoint objects."
    
    There's not necessarily a 1:1 relation between the check source node
    name and the endpoint name.
    
    This reverts commit a26df23e3d55abee4d0bd3c34da49d63702d209f.
    
    Refs #5810
    Refs #5636
    
    Signed-off-by: Michael Friedrich <michael.friedrich@netways.de>


---

**Parent Task:** [#5636](https://dev.icinga.com/issues/5636)
